### PR TITLE
Implement Placeholder in IEntry Handlers

### PR DESCRIPTION
**Pull request by almirvuk**
Implements #374 

- Adds string Placeholder { get; } to the IEntry interface
- Adds Placeholder property map to EntryHandler
- Adds Placeholder mapping methods to EntryHandler for Android and iOS
- Adds extension methods to apply Placeholder on Android/iOS
- Adds DeviceTests for initial Placeholder values on iOS and Android

### CODE REVIEW

File: src/Core/src/Platform/Android/EntryExtensions.cs [hunk 1]
```diff
@@ -51,5 +51,13 @@ internal static void SetInputType(this EditText editText, IEntry entry)
 			if (entry.IsPassword && ((editText.InputType & InputTypes.ClassNumber) == InputTypes.ClassNumber))
 				editText.InputType |= InputTypes.NumberVariationPassword;
 		}
+
+		public static void UpdatePlaceholder(this EditText editText, IEntry entry)
+		{
+			if (editText.Hint == entry.Placeholder)
```
Comment (mattleibow) [inline src/Core/src/Platform/Android/EntryExtensions.cs:57]: Is this really worth the extra interop. If we are different, then we have to set. If we are the same, then it is equivalent to setting it.

**Review by mattleibow — COMMENTED** [review]
Thanks for the PR. Looks really good.

Before I approve, I just want a thought from the team about the placeholder color. I see in core forms, we have a `IPlaceholderElement` that has the text and color. Not sure if we are going to do something like that? Do we also want our own `interface IPlaceholderText : IText` or something?

Also, may be related, I see we have an `ITextInput` that is currently unused. Maybe that is what we want and we can remove the properties we have not yet implemented?

File: src/Core/src/Platform/Android/EntryExtensions.cs (hunk 1 above)
Comment (mattleibow) [reply src/Core/src/Platform/Android/EntryExtensions.cs:57]: I see you added this in a special commit. What was the reason for this? What do you know that I don't? XD

Comment (almirvuk) [reply src/Core/src/Platform/Android/EntryExtensions.cs:57]: I seen this in EntryRenderer for XF 😀 and it made sense to me to not assign them if they already have same value.

Comment (mattleibow) [reply src/Core/src/Platform/Android/EntryExtensions.cs:57]: OK, let's leave it as the renderer and be safe.

**Comment by jsuarezruiz** [comment]
I think Placeholder property have sense in `ITextInput`:

```
public interface ITextInput : IText
{
     string Placeholder { get; }
     Color PlaceholderColor { get; }
}
```

Then, use the interface in different controls: Entry, Editor, etc.

**Comment by almirvuk** [comment]
@jsuarezruiz So I should implement ITextInput in Entry and use the property from the interface? :)

**Comment by mattleibow** [comment]
Keep the color for a new PR I think. Not sure if we want to make it a Brush or something, so once we decide exactly how colors work, we can add.

**Comment by almirvuk** [comment]
@jsuarezruiz replaced property with usage of ITextInput interface :) Can you review it now, please :)

**Review by jsuarezruiz — DISMISSED** [review]
Looks nice, just a detail, could you use the [PortHandler] attribute to mark in the Renderers that the code to update the Placeholder has been ported?

**Comment by almirvuk** [comment]
Sure, I will add them!

**Comment by mattleibow** [comment]
> Looks nice, just a detail, could you use the [PortHandler] attribute to mark in the Renderers that the code to update the Placeholder has been ported?

we need to add this to the wiki and the PR @rachelkang is working on

File: src/Core/src/Core/IEntry.cs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿namespace Microsoft.Maui
 {
-	public interface IEntry : IView, IText
+	public interface IEntry : IView, IText, ITextInput
```
Comment (mattleibow) [inline src/Core/src/Core/IEntry.cs:3]: I think `ITextInput` has a bunch of props, so maybe we might have to delete them all for now except the placeholder?

Comment (almirvuk) [reply src/Core/src/Core/IEntry.cs:3]: I would leave them for now, because other handlers are coming in a few days :)

Comment (mattleibow) [reply src/Core/src/Core/IEntry.cs:3]: But will it compile without being implemented in the stubs?

Also, we do want to remove the color property entirely as we will use a brush or something.

Comment (almirvuk) [reply src/Core/src/Core/IEntry.cs:3]: With the last commit, I implemented them in the Entry stub.

"Also, we do want to remove the color property entirely as we will use a brush or something." - Ok, then I will do based on suggestions when we figure out what to do with ITextInput :)

File: src/Core/tests/DeviceTests/Stubs/EntryStub.cs [hunk 3]
```diff
@@ -13,5 +13,15 @@ public partial class EntryStub : StubBase, IEntry
...
+
+		public Keyboard Keyboard { get; set; }
+
+		public bool IsSpellCheckEnabled { get; set; }
+
+		public int MaxLength { get; set; }
+
+		public Color PlaceholderColor { get; set; }
+
+		public bool IsReadOnly { get; set; }
```
Comment (mattleibow) [inline src/Core/tests/DeviceTests/Stubs/EntryStub.cs:25]: Hmmmm. I think we need to remove these for now? @rmarinho @PureWeen

We can always add it back. Not sure if we are going to add all these props. @hartez did you have issues for each, or are there any that we are not going to do?

File: src/Core/tests/DeviceTests/Stubs/EntryStub.cs [hunk 4]
```diff
@@ -9,7 +9,19 @@ public partial class EntryStub : StubBase, IEntry
...
+
 		public bool IsTextPredictionEnabled { get; set; }
+
+		public Keyboard Keyboard { get; set; }
+
+		public bool IsSpellCheckEnabled { get; set; }
+
+		public int MaxLength { get; set; }
+
+		public Color PlaceholderColor { get; set; }
```
Comment (mattleibow) [inline src/Core/tests/DeviceTests/Stubs/EntryStub.cs:23]:
```suggestion
```

File: src/Core/src/Core/IEntry.cs (hunk 2 above)
Comment (mattleibow) [reply src/Core/src/Core/IEntry.cs:3]: Let's do this in a new PR. Then we can merge this sooner :)

Comment (almirvuk) [reply src/Core/src/Core/IEntry.cs]: I removed the usage of ITextInput, when we have a clear clue what to do with that interface we can add it later, now currently the Placeholder is only property (not coming from the interface). Hope now it is ready for merge 😄

**Review by mattleibow — APPROVED** [review]
Looks good enough to merge!

**Review by mattleibow — APPROVED** [review]

**Comment by almirvuk** [comment]
Another one! Yeah!!! Thank you for the time! 🎊

**Comment by mattleibow** [comment]
Thanks for the work!

**Pull request merged**